src/components/Cards/WeatherCard.tsx

```diff
@@ -112,12 +112,13 @@ const UVIndex = (): JSX.Element => {
     const theme = useTheme();
     const { VITE_ZIP_CODE, VITE_TIME_INTERVAL } = import.meta.env;
     const uvAPIURL = `https://data.epa.gov/efservice/getEnvirofactsUVHOURLY/ZIP/${VITE_ZIP_CODE}/JSON`;
-    const uvIndexData: any[] | undefined = callExternalAPIOnInterval(VITE_TIME_INTERVAL, uvAPIURL);
-
+    let uvIndexData: any[] | undefined = callExternalAPIOnInterval(VITE_TIME_INTERVAL, uvAPIURL);
+    uvIndexData = Array.isArray(uvIndexData) ? uvIndexData : [];
     const currentHour = DateTime.now().toFormat("hh a");
+
     const uvIndex =
-        uvIndexData?.find((data: { DATE_TIME: string }) => data.DATE_TIME.includes(currentHour))
-            ?.UV_VALUE || "-";
```

Comment:
Let's keep this question mark, as it's important to prevent the whole midnight crash thing.